dot/peerset/peerset.go

```diff
@@ -645,18 +670,21 @@ func (ps *PeerSet) disconnect(setIdx int, reason DropReason, peers ...peer.ID) e
 		state.nodes[pid] = n
 
 		if err = state.disconnect(setIdx, pid); err != nil {
-			return err
+			return fmt.Errorf("cannot disconnect peer %s at set %d: %w", pid, setIdx, err)
 		}
-		ps.resultMsgCh <- Message{
+
+		dropMessage := Message{
 			Status: Drop,
 			setID:  uint64(setIdx),
 			PeerID: pid,
 		}
 
+		ps.processor.Process(dropMessage)
+
 		// TODO: figure out the condition of connection refuse.
```

Comment:
is this still a todo and/or is there an issue for it?

Reply:
I'm not sure, @noot do you know if is this still a todo?